Return type for Azure funcs

Abrhm7786 opened the pull request:
Change to Azure functions code gen which gives the functions an explicit return type instead of IActionResult.

<!-- Please check the completed items below -->
### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

aqwhunter commented:
@Abrhm7786 , would you please clarify?  Has the called method been changed elsewhere, namely its returned type.  If it returns IActionResult, how can we use a local type return?  Would `ActionResult<T>` be an appropriate result?

Abrhm7786 commented:
> @Abrhm7786 , would you please clarify? Has the called method been changed elsewhere, namely its returned type. If it returns IActionResult, how can we use a local type return? Would `ActionResult<T>` be an appropriate result?

Thanks for this. I tested and found that Task<IActionResult<T> works best, so I have changed it to that. Also updated the readme file.

wing328 commented on docs/generators/csharp-netcore-functions.md:
```diff
@@ -1,7 +1,9 @@
 ---
+## DESCRIPTION
 
+Creates Azure function templates on top of the models/converters created by the C# codegens. This function is contained in a partial class. Default Get/Create/Patch/Post etc. methods are created with an underscore prefix. The assumption is that when the function is implemented, the partial class will be completed with another partial class. The implementing code should be located in a method of t ...
```
this file is auto-generated so I'll add these to the getHelp() section of the generator instead after merging this PR.

wing328 replied:
Filed https://github.com/OpenAPITools/openapi-generator/pull/12183
